js/runner.go

```diff
@@ -356,65 +353,46 @@ func (r *Runner) timeoutErrorDuration(stage string) time.Duration {
 type VU struct {
 	BundleInstance
 
-	Runner    *Runner
-	Transport *http.Transport
-	Dialer    *netext.Dialer
-	CookieJar *cookiejar.Jar
-	TLSConfig *tls.Config
-	ID        int64
-	Iteration int64
+	Runner     *Runner
+	RunContext *context.Context
+	Transport  *http.Transport
+	Dialer     *netext.Dialer
+	CookieJar  *cookiejar.Jar
+	TLSConfig  *tls.Config
+	ID         int64
+	Iteration  int64
 
 	Console *console
 	BPool   *bpool.BufferPool
 
 	Samples chan<- stats.SampleContainer
 
 	setupData goja.Value
-
-	// A VU will track the last context it was called with for cancellation.
-	// Note that interruptTrackedCtx is the context that is currently being tracked, while
-	// interruptCancel cancels an unrelated context that terminates the tracking goroutine
-	// without triggering an interrupt (for if the context changes).
-	// There are cleaner ways of handling the interruption problem, but this is a hot path that
-	// needs to be called thousands of times per second, which rules out anything that spawns a
-	// goroutine per call.
-	interruptTrackedCtx context.Context
-	interruptCancel     context.CancelFunc
-
-	m *sync.Mutex
 }
 
-// Verify that VU implements lib.VU
-var _ lib.VU = &VU{}
+// Verify that interfaces are implemented
+var _ lib.ActiveVU = &VU{}
+var _ lib.InitializedVU = &VU{}
```

Comment:
This was a strange choice that I also didn't quite understand from the previous attempt at this refactoring - why would you want the same `VU` struct to implement both interfaces? It seems much cleaner to me to use a separate `ActiveVU` struct that contains the `VU`, but also the `lib.VUActivationParams`.

Reply:
I considered that, but found it simpler to reuse the same struct. Right now it's handy to convert the same `VU` to both interfaces, whereas a bit more wrangling would be needed otherwise.

But sure, your suggestion sounds good, I'll see how it works.

Reply:
Maybe resolved by ea258f89?

You're right, I like the separation better. :+1: